plugins/home/src/homePageComponents/FeaturedDocs/FeaturedDocs.tsx

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import useAsync from 'react-use/lib/useAsync';
+import {
+  LinkButton,
+  EmptyState,
+  Link,
+  CodeSnippet,
+  InfoCard,
+  Progress,
+  WarningPanel,
+} from '@backstage/core-components';
+import { catalogApiRef, CatalogApi } from '@backstage/plugin-catalog-react';
+import { useApi } from '@backstage/core-plugin-api';
+import { EntityFilterQuery } from '@backstage/catalog-client';
+import { ClassNameMap } from '@material-ui/styles';
+
+import { makeStyles, Theme, Typography } from '@material-ui/core';
+
+type DocsCardProps = {
+  filter: EntityFilterQuery;
+  color?: 'inherit' | 'primary' | 'secondary' | undefined;
+  customStyles?: ClassNameMap<string> | undefined;
+  emptyState?: React.ReactNode | undefined;
+  subLinkText?: string | undefined;
+  title?: React.ReactNode | string | undefined;
+};
+
+const useStyles = makeStyles<Theme>(() => ({
+  docDescription: {
+    fontSize: 16,
+    fontWeight: 400,
+    marginBottom: '16px',
+    marginTop: '12px',
+  },
+  docSubLink: {
+    fontSize: 12,
+    fontWeight: 700,
+    lineHeight: 2,
+  },
+  docsTitleLink: {
+    fontSize: 16,
+    fontWeight: 600,
+    lineHeight: 2,
+  },
+}));
+
+/**
+ * A component to display specific Featured Docs.
+ * @param {EntityFilterQuery} filter - The entity filter used to display only the intended item/s
```

Comment:
you shouldn't repeat all the types here - no need - typescript extracts them for us. We just do `@param name - desc`.

Even better - if you don't put any params here at all, but instead put those doc strings on individual fields inside `DocsCardProps`, then that type gets fully documented instead which is more correct. Because actually, if you think about it, `FeaturedDocs` has only one single param, namely `props`